### PR TITLE
Fix multi-line string edit formatter problem

### LINKED ISSUE #2270: Formatter problem with multi-line strings
I was writing a multi-line `SQL` as a `def` and found an interesting issue. I have turned off all extensions except for `Calva` and removed the contents of my `cljfmt.edn` to simplify and it still happens. I wasn't able to repro this by calling `cljfmt` from a prompt.

To repro, start with a file like

```clojure
(ns a-project.repro)

(def a-def "
            
            ")

(defn a-function [])
```

Put your cursor on the end of the blank line after `a-def`. Type a letter and then format the file and it turns into

```clojure
(ns little-differ.repro)

(def a-def "
a
")

(defn a-function [])"
```

Notice that an extra `"` is added to the end of the file after the last `)`. If I take out the newlines at the beginning and end of my string then no problems happen. Any ideas? Happy to work on a fix if people have ideas what I should do.

### PR DESCRIPTION
**Pull request by bpringe**
## What has changed?

- Pass `startState` to `scanner.processLine` in the constructor of `TextLine`

Fixes #2270 

## My Calva PR Checklist

I have:

- [x] Read [How to Contribute](https://github.com/BetterThanTomorrow/calva/wiki/How-to-Contribute#before-sending-pull-requests).
- [x] Directed this pull request at the `dev` branch. (Or have specific reasons to target some other branch.)
- [x] Made sure I have changed the PR base branch, so that it is not `published`. (Sorry for the nagging.)
- [x] Made sure there is an issue registered with a clear problem statement that this PR addresses, (created the issue if it was not present).
    - [x] Updated the `[Unreleased]` entry in `CHANGELOG.md`, linking the issue(s) that the PR is addressing.
    - [x] Referenced the issue I am fixing/addressing _in a commit message for the pull request_.
        - [x] If I am fixing the issue, I have used [GitHub's fixes/closes syntax](https://help.github.com/en/articles/closing-issues-using-keywords)
        - [x] If I am fixing just part of the issue, I have just referenced it w/o any of the "fixes” keywords.
- [x] Figured if **anything** about the fix warrants tests on Mac/Linux/Windows/Remote/Whatever, and either tested it there if so, or mentioned it in the PR.
- [x] Added to or updated docs in this branch, if appropriate
- [x] Tests
  - [x] Tested the particular change
  - [x] Figured if the change might have some side effects and tested those as well.
- [x] Formatted all JavaScript and TypeScript code that was changed. (use the [prettier extension](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) or run `npm run prettier-format`)
- [x] Confirmed that there are no linter warnings or errors (use the [eslint extension](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint), run `npm run eslint` before creating your PR, or run `npm run eslint-watch` to eslint as you go).

Ping @pez, @bpringe, @corasaurus-hex, @Cyrik

### CODE REVIEW

File: src/extension-test/unit/cursor-doc/paredit-test.ts [hunk 1]
```diff
@@ -1662,9 +1662,7 @@ describe('paredit', () => {
         expect(textAndSelection(a)).toEqual(textAndSelection(b));
       });
 
-      // NB: enabling this breaks bunch of other tests.
-      //     Not sure why, but it can be run successfully by itself.
-      xit('splice string', async () => {
+      it('splice string', async () => {
```
Comment (PEZ) [inline src/extension-test/unit/cursor-doc/paredit-test.ts:1665]: This is plain wonderful to be able to enable again. It has bugged me to no end why one test would mess up for other tests, but your fix makes it obvious why that could happen.  Thanks, @bpringe! ❤️ 🎉

**Comment by PEZ** [comment]
Thanks! ❤️ 🙏  So sweet. You found the bug that I have been wondering about for several years now. 🎉 I've been looking in the wrong place!

File: README.md [hunk 2]
```diff
@@ -2,29 +2,30 @@
...
-* [structural editing](https://calva.io/paredit/)
-* code [formatting](https://calva.io/formatting/)
-* code navigation
-* a [debugger](https://calva.io/debugger/)
-* [linting](https://calva.io/linting/)
-* syntax highlighting
-* Clojure aware rainbow brackets
-* a [test runner](https://calva.io/test-runner/)
-* refactoring support
-* and much more.
```
Comment (PEZ) [inline README.md]: I don't fancy the bulk change to the README about the bullets, though. Generally I use asterisk  as bullets and the Markdown spec clearly allows it. Can we discuss it separately?

File: src/extension-test/unit/cursor-doc/paredit-test.ts (hunk 1 above)
Comment (bpringe) [reply src/extension-test/unit/cursor-doc/paredit-test.ts:1665]: Awesome! Glad to help.

File: README.md (hunk 2 above)
Comment (bpringe) [reply README.md]: Those changes actually happened due to formatting on save with the prettier extension enabled, it seems. I wasn't sure at the time what exactly did the formatting, but I just noticed the formatting doesn't happen with prettier disabled. I reverted the changes.

Comment (PEZ) [reply README.md]: Interesting. I wonder if we may have different prettier settings. Let me check if I have something in my user settings about it...

Comment (PEZ) [reply README.md]: No, I can't find anything there. In our package.json we define these scripts:

```json
    "prettier-format": "npx prettier --write \"./**/*.{ts,js,json}\"",
    "prettier-check": "npx prettier --check \"./**/*.{ts,js,json}\"",
    "prettier-check-watch": "onchange \"./**/*.{ts,js,json}\" -- prettier --check {{changed}}",
    "prettier-format-watch": "onchange \"./**/*.{ts,js,json}\" -- prettier --write {{changed}}",
    "preprettier-format-watch": "npm run prettier-format",
```

So it shouldn't format markdown from those at least. Maybe you have some setting making the extension do that stuff?

Comment (bpringe) [reply README.md]: It happens with the prettier extension enabled if I run "Format Document" on the readme and have prettier set as my formatter for markdown.

```json
  "[markdown]": {
    "editor.defaultFormatter": "esbenp.prettier-vscode"
  }
```

**Pull request merged**